frontend/src/components/Navbar.js

```diff
@@ -0,0 +1,139 @@
+import * as React from 'react';
+import {
+  AppBar,
+  Box,
+  Toolbar,
+  IconButton,
+  Typography,
+  Menu,
+  Container,
+  Avatar,
+  Button,
+  MenuItem,
+  Icon
+} from '@mui/material'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faBars } from '@fortawesome/free-solid-svg-icons'
+
+
+
+const settings = ['Account', 'Logout'];
+
+const Navbar = () => {
+  const [anchorElUser, setAnchorElUser] = React.useState(null);
+
+  const [anchorElWorkspaces, setAnchorElWorkspaces] = React.useState(null);
+  const [anchorElRecent, setAnchorElRecent] = React.useState(null);
+  const [anchorElActions, setAnchorElActions] = React.useState(null);
+
+
+  const handleClose = () => {
+    setAnchorElWorkspaces(null);
+    setAnchorElRecent(null);
+    setAnchorElActions(null);
+  };
```

Comment:
Czy istnieje jakiś lepszy sposób na rozwiązanie logiki rozwijanych menu?
```
            <Button
                id="actions-button"
                onClick={e => setAnchorElActions(e.currentTarget)}
                sx={{mr:4}}
            >
                <Typography variant="button" color="common.white">Actions</Typography>
            </Button>
            <Menu
                sx={{ mt: 2 }}
                id="actions-button"
                anchorEl={anchorElActions}
                open={Boolean(anchorElActions)}
                onClose={e => setAnchorElActions()}
            >
                <MenuItem onClick={handleClose}>aaaaaaaaaaaaaaaaaaaaa</MenuItem>
                <MenuItem onClick={handleClose}>aaaaaaaaaaaaaaaaaaaaa</MenuItem>
                <MenuItem onClick={handleClose}>aaaaaaaaaaaaaaaaaaaaa</MenuItem>
            </Menu>
```

Reply:
Pomysł jest ok. Na pewno zrobiłbym z tego osobny komponent w stylu `DropdownButton`, bo masz kilka takich przycisków z rozwijanym Menu i zawsze będą potrzebowały `Button` + `Menu` z odpowienio podpiętymi eventami, żeby działać.

Reply:
Wtedy unikniesz duplikacji 4 razy tego samego kodu i o wiele łatwiej będzie tym zarządzać

Reply:
W React (w przeciwieństwie do nieludzkiej figmy) możesz przekazać content do swojego komponentu.

Mogłoby to tak wyglądać

```js
// frontend/src/components/DropdownButton.js
const DropdownButton = (props) => {
    ...
    return (
        // skrót od `<div>`
        <>
            <Button
                id="actions-button"
                onClick={e => setAnchorElActions(e.currentTarget)}
                sx={{mr:4}}
            >
                <Typography variant="button" color="common.white">props.label</Typography>
            </Button>
            <Menu
                sx={{ mt: 2 }}
                id="actions-button"
                anchorEl={anchorElActions}
                open={Boolean(anchorElActions)}
                onClose={e => setAnchorElActions()}
            >
                {/* przekaż elementy 'dzieci' otrzymane w `DropdownButton` do `Menu` */}
                {props.children}
            </Menu>
        </>
    )
}
```

```js
// fronted/src/components/Navbar.js
const Navbar = () => {
  return (
    ...
    <DropdownButton label="Actions">
        {/* Jako 'dzieci' przekazujesz elementy Menu */}
        <MenuItem onClick={someAction}>aaaaaaaaaaaaaaaaaaaaa</MenuItem>
        <MenuItem onClick={someAction}>aaaaaaaaaaaaaaaaaaaaa</MenuItem>
        <MenuItem onClick={someAction}>aaaaaaaaaaaaaaaaaaaaa</MenuItem>
    </DropdownButton>
    ...
  )
}
```